### PR TITLE
Sema: warn when POSITION[0] is used intead of SV_Position

### LINKED ISSUE #3742: No warning/error when SV_Position semantic is missing from Vert/Domain/Geom output and POSITION is used instead
### Title
No warning/error when SV_Position semantic is missing from Vert/Domain/Geom output and POSITION is used instead

### Functional impact
Shaders written for FXC which use the d3d9 style POSITION semantic for the vertex pipeline output will compile with DXC but will produce no output rendering. The POSITION semantic is just interpreted as a user semantic but the fact that SV_Position is not written out is not reported, which is most often a user error. P.S. this gotcha is also not documented in the [Porting shaders from FXC to DXC](https://github.com/microsoft/DirectXShaderCompiler/wiki/Porting-shaders-from-FXC-to-DXC) wiki

### Minimal repro steps
1. Compile the following shader with `dxc.exe -T vs_6_0 -E Vert repro_shader.txt` (or `with dxc.exe -T vs_6_0 -E Vert -spirv repro_shader.txt` for the SPIR-V backend)
```
float4 Vert(float4 position : POS) : POSITION
{
	return position;
}
```

### Expected result
An error or at least a warning is printed when there is no SV_Position in the Vertex/Domain/Geometry shader output, possibly pointing to the user defined POSITION semantic as the place for user correction. Similar to what DXC does for uses of the legacy COLOR or DEPTH semantics in the pixel shader output.

### Actual result
Shader compiles without any `POS` SysValue (for the DXIL backend) and no `gl_Position` (for the SPIR-V backend) in the output. The resulting shader produces no rendering at runtime.

### Further technical details
**N.B.** To my understanding DXC should probably only not warn/error when using the stream output stage in the vertex pipeline but we at Unity don't ever use stream output and would prefer if DXC had a compiler flag to enable this warning when we know before compilation that the shader will never be used for stream output.
Used DXC release v1.6.2104 (e09a454)

### PR DESCRIPTION
**Pull request by Keenuts**
Before SM4, POSITION was equivalent to today's SV_Position. However, as HLSL moved to system semantics (SV_), the support for those old semantics has been deprecated.
Some like VFACE are now completely forbidden, but others like POSITION or COLOR have been handled differently by FXC and DXC.

Today, usage of POSITION in place of SV_Position is probably a mistake as POSITION is now considered to be a user-defined semantic.

This commit adds a warning when POSITION is used as an output semantic for a vertex shader, as most cases should be the result of a mistake. If this is expected, the warning can be disabled by using `-Wno-dx9-deprecation`.

Fixes #3742

### CODE REVIEW

**Comment by Keenuts** [comment]
@pow2clk @tex3d

File: tools/clang/include/clang/Basic/DiagnosticSemaKinds.td
Comment (sudonatalie) [inline tools/clang/include/clang/Basic/DiagnosticSemaKinds.td:7716]: optional: I think a slightly more idiomatic clang phrasing might be: `'POSITION' is a user-defined semantic; did you mean 'SV_Position'?`

Comment (Keenuts) [reply tools/clang/include/clang/Basic/DiagnosticSemaKinds.td]: Yes, way better message!

File: tools/clang/lib/Sema/SemaHLSL.cpp
Comment (Keenuts) [reply tools/clang/lib/Sema/SemaHLSL.cpp:15303]: Yes, that's better, done

**Review by sudonatalie — APPROVED** [review]
LGTM pending approval from someone at MS

**Comment by Keenuts** [comment]
@llvm-beanz ping? (or @tex3d?)

**Pull request merged**
